sdk/samples/iot/aziot_realtek_amebaD/aziot_realtek_amebaD.ino

```diff
@@ -0,0 +1,385 @@
+  }
+
+  Serial.println("");
+  Serial.print("WiFi connected, IP address: ");
+  Serial.println(WiFi.localIP());
+}
+
+void receivedCallback(char* topic, byte* payload, unsigned int length)
+{
+  Serial.print("Received [");
+  Serial.print(topic);
+  Serial.print("]: ");
+  for (unsigned int i = 0; i < length; i++)
+  {
+    Serial.print((char)payload[i]);
+  }
+}
+
+static void initializeClients()
+{
+  Serial.println("Initializing MQTT client");
+  
+  wifi_client.setRootCA(ca_pem_nullterm);
+
+  if (az_result_failed(az_iot_hub_client_init(
+          &hub_client,
+          az_span_create((uint8_t*)host, strlen(host)),
+          az_span_create((uint8_t*)device_id, strlen(device_id)),
+          NULL)))
+  {
+    Serial.println("Failed initializing Azure IoT Hub client");
+    return;
+  }
+
+  mqtt_client.setServer(host, port);
+  mqtt_client.setCallback(receivedCallback);
+
+  Serial.println("MQTT client initialized");
+}
+
+int64_t iot_sample_get_epoch_expiration_time_from_minutes(uint32_t minutes)
+{
+
+  int64_t expiration_time = 0;
+  long ts = 0;
+  long tus = 0;
+  unsigned int ttk = 0;
+
+  //it should be ok to do init more than one times. It'll handle inside sntp_init().
+  sntp_init();
+
+  sntp_get_lasttime(&ts,&tus,&ttk);
+  
+  while(ts == 0){
+    vTaskDelay(1000 / portTICK_RATE_MS);
+    sntp_get_lasttime(&ts,&tus,&ttk);
+  } 
+
+  expiration_time = (int64_t)ts + minutes * 60;
+  
+  return expiration_time;
+}
+
+static void hmac_sha256_sign_signature(
+    az_span decoded_key,
+    az_span signature,
+    az_span signed_signature,
+    az_span* out_signed_signature)
+{ 
+  if(rom_hmac_sha256(
+    az_span_ptr(decoded_key), 
+    (size_t)az_span_size(decoded_key),
+    az_span_ptr(signature),
+    (size_t)az_span_size(signature),
+    az_span_ptr(signed_signature)) != 0)
+
+  {
+    Serial.println("[ERROR] rom_hmac_sha256 failed");
+  };
+
+  *out_signed_signature = az_span_create(az_span_ptr(signed_signature), 32);
+}
+
+static void base64_encode_bytes(
+    az_span decoded_bytes,
+    az_span base64_encoded_bytes,
+    az_span* out_base64_encoded_bytes)
+{
+  size_t len;
+  if( mbedtls_base64_encode( az_span_ptr(base64_encoded_bytes), (size_t)az_span_size(base64_encoded_bytes), 
+    &len, az_span_ptr(decoded_bytes), (size_t)az_span_size(decoded_bytes)) != 0)
+  {
+    Serial.println("[ERROR] mbedtls_base64_encode fail");
+  }
+  
+  *out_base64_encoded_bytes = az_span_create(az_span_ptr(base64_encoded_bytes), (int32_t)len);
+}
+
+static void decode_base64_bytes(
+    az_span base64_encoded_bytes,
+    az_span decoded_bytes,
+    az_span* out_decoded_bytes)
+{
+  
+  memset(az_span_ptr(decoded_bytes), 0, (size_t)az_span_size(decoded_bytes));
+
+  size_t len;
+  if( mbedtls_base64_decode( az_span_ptr(decoded_bytes), (size_t)az_span_size(decoded_bytes), 
+      &len, az_span_ptr(base64_encoded_bytes), (size_t)az_span_size(base64_encoded_bytes)) != 0)
+  {
+    Serial.println("[ERROR] mbedtls_base64_decode fail");
```

Comment:
I will keep as is because that's directly copied from the Realtek repo. In case in the future we use their functions directly, our usage won't differ and will require no changes from our side to consume their code.